chainer/distributions/kl_divergence.py

```diff
@@ -0,0 +1,69 @@
+from chainer.distributions import normal
+from chainer.functions.math import exponential
+
+_KLDIVERGENCE = {}
+
+
+def register_kl(dist1, dist2):
+    """Decorator to register KL divergence function.
```

Comment:
Write when/how to use this decorator and what can be done with it. Hint:
- What should be the signature of the decorated function?
- The decorated function will be called by `kl_divergence` based on the argument types. Using it, we can write distribution-agnostic code to compute KL divergence.